dallinger/heroku/messages.py

```diff
@@ -0,0 +1,190 @@
+process, implying that the assignment had not been successfully completed. Of
+course when the seriousness of this scenario dawned on me I had to depend on
+my trusting walking stick for support: without the notification I didn't know
+to remove the old assignment's data from the tables and AWS will have already
+sent their replacement, meaning that the tables may already be in a most
+unsound state!
+
+I am sorry to trouble you with this, however, I do not know how to proceed so
+rather than trying to remedy the scenario myself, I have instead temporarily
+ceased operations by expiring the HIT with the fellows at AWS and have
+refrained from posting any further invitations myself. Once you see fit I
+would be most appreciative if you could attend to this issue with the caution,
+sensitivity and intelligence for which I know you so well.
+
+I remain your faithful and
+obedient servant,
+William H. Dallinger
+
+P.S. Please do not respond to this
+message, I am busy with other matters.
+"""
+
+cancelled_hit_nonwhimsical = """Dear experimenter,
+
+This is an automated email from
+Dallinger. You are receiving this email because the Dallinger platform has
+discovered evidence that a notification from Amazon Web Services failed to
+arrive at the server. Dallinger has automatically contacted AWS and has
+determined the dropped notification was an abandoned/returned notification
+(i.e. the participant had returned the experiment or had run out of time).
+This is a serious error and so Dallinger has paused the experiment - expiring
+the HIT on MTurk and setting auto_recruit to false. Participants currently
+playing will be able to finish, however no further participants will be
+recruited until you do so manually. We strongly suggest you use the details
+below to check the database to make sure the missing notification has not caused
+additional problems before resuming.
+If you are receiving a lot of these
+emails this suggests something is wrong with your experiment code.
+
+Best,
+
+The Dallinger dev. team.
+
+Error details:
+Assignment: {assignment_id}
+
+Allowed time: {duration}
+Time since participant started: {minutes_so_far}
+"""
+
+
+class EmailingHITMessager(object):
+
+    def __init__(self, when, assignment_id, hit_duration, time_active, config, server=None):
+        self.when = when
+        self.assignment_id = assignment_id
+        self.duration = round(hit_duration / 60)
+        self.minutes_so_far = round(time_active / 60)
+        self.minutes_excess = round((time_active - hit_duration) / 60)
+        self.whimsical = config.get("whimsical")
+        self.username = config.get('dallinger_email_username')
+        self.fromaddr = self.username + "@gmail.com"
+        self.toaddr = config.get('contact_email_on_error')
+        self.email_password = config.get("dallinger_email_key")
+        self.server = server or SMTP('smtp.gmail.com:587')
+
+    def _send(self, data):
```

Comment:
This comment is probably out-of-scope for this PR given that it's focused on refactoring what's already there, but @thomasmorgan opened #171 a while back, reporting that Gmail security settings prevent using a Heroku dyno to send email from a Gmail address. If you happen to know a way around that, this might be a good time fix the problem. Otherwise, I'll leave the issue open and we can fix it some other time.